acme/src/certificate.rs

```diff
@@ -49,13 +62,24 @@ impl RustyAcme {
 
                 // only verify that leaf has the right identity fields
                 if i == 0 {
-                    Self::verify_leaf_certificate(cert, &order.try_get_coalesce_identifier()?, hash_alg, env)?;
+                    Self::verify_leaf_certificate(cert, &order.try_get_coalesce_identifier()?, hash_alg, env.as_ref())?;
                 }
                 acc.push(cert_pem.contents().to_vec());
                 Ok(acc)
             })
     }
 
+    fn extract_intermediates(mut pems: Vec<pem::Pem>) -> RustyAcmeResult<Vec<Certificate>> {
+        use x509_cert::der::Decode as _;
+        pems.remove(0);
+        pems.iter()
+            .try_fold(vec![], |mut acc, pem| -> RustyAcmeResult<Vec<Certificate>> {
+                let cert = x509_cert::Certificate::from_der(pem.contents())?;
+                acc.push(cert);
+                Ok(acc)
+            })
+    }
```

Comment:
you can write this function to take a reference and start the iter from the second element. Calling `iter().skip(1)` should to the trick

Reply:
And since I'm no longer mutating the array I could also change the `pems` argument to a reference.